File: generate_template.js
const XLSX = require('xlsx');

// Função para criar o workbook
function createTemplate(filename, isHome) {
    const matchInfo = [
        ["Data", "AAAA-MM-DD"],
        ["Local", "Escreva o Local"],
        ["Competicao", "Nome da Competição"],
        ["Jornada", "Fase Regular"],
        ["Hora", "HH:MM"],
        ["Equipa Casa", "Nome da Equipa Casa"],
        ["Equipa Adversaria", "Nome da Equipa Adversária"],
        ["Sigla Casa", "Sigla Casa (ex: SLB)"],
        ["Sigla Adversaria", "Sigla Adv (ex: SCP)"],
        ["Equipa Analisada", isHome ? "Nome da Equipa Casa" : "Nome da Equipa Adversária"]
    ];

    // Players
    const players = [
        [1, 1, "Nome", "Apelido", "GR"],
        [2, 2, "Nome", "Apelido", "FIXO"],
        [3, 3, "Nome", "Apelido", "ALA"],
        [4, 4, "Nome", "Apelido", "ALA"],
        [5, 5, "Nome", "Apelido", "PIVOT"],
        [6, 6, "Nome", "Apelido", "PIVOT"],
        [7, 7, "Nome", "Apelido", "ALA"],
        [8, 8, "Nome", "Apelido", "ALA"],
        [9, 12, "Nome", "Apelido", "GR"],
        [10, 10, "Nome", "Apelido", "ALA"],
        [11, 11, "Nome", "Apelido", "ALA"],
        [12, 13, "Nome", "Apelido", "FIXO"],
        [13, 14, "Nome", "Apelido", "PIVOT"],
        [14, 15, "Nome", "Apelido", "ALA"],
        [15, 16, "Nome", "Apelido", "ALA"],
        [16, 17, "Nome", "Apelido", "FIXO"]
    ];

    const ws_data = [];
    matchInfo.forEach(row => ws_data.push(row));
    while (ws_data.length < 12) {
        ws_data.push([]);
    }
    ws_data.push(["Player ID", "Number", "First Name", "Last Name", "Position"]);
    players.forEach(p => ws_data.push(p));

    const wb = XLSX.utils.book_new();
    const ws = XLSX.utils.aoa_to_sheet(ws_data);
    XLSX.utils.book_append_sheet(wb, ws, "Setup");
    XLSX.writeFile(wb, filename);
    console.log(`${filename} created successfully.`);
}

createTemplate("Template_Equipa_Casa.xlsx", true);
createTemplate("Template_Equipa_Visitante.xlsx", false);

File: generate_template.py
import pandas as pd
from openpyxl import Workbook
from openpyxl.styles import Font, PatternFill

# Create a new workbook
wb = Workbook()
ws = wb.active
ws.title = "Setup"

# --- MATCH INFO (Rows 1-10) ---
# Column A must contain these exact keys for the parser to work
match_keys = [
    "Match Date", "Location", "Competition", "Stage", 
    "Kick Off Time", "Home Team", "Away Team", 
    "Home Display", "Away Display", "Team Analyzed"
]
# Column B: Sample Values (User edits these)
match_values = [
    "2024-02-01", "Pavilhão", "Liga", "Jornada 1", 
    "16:00", "Equipa Casa", "Equipa Fora", 
    "CASA", "FORA", "Equipa Casa"
]

for i, (key, val) in enumerate(zip(match_keys, match_values), start=1):
    ws.cell(row=i, column=1, value=key).font = Font(bold=True)
    ws.cell(row=i, column=2, value=val)

# --- PLAYERS (Rows 14-29) ---
# Header at Row 13 for reference
headers = ["Player ID", "Number", "First Name", "Last Name", "Position"]
for col, h in enumerate(headers, start=1):
    cell = ws.cell(row=13, column=col, value=h)
    cell.font = Font(bold=True)
    cell.fill = PatternFill(start_color="DDDDDD", end_color="DDDDDD", fill_type="solid")

# Sample Players (16 slots)
# The app reads 16 rows starting at 14.
players = [
    [1, 1, "João", "Silva", "GK"],
    [2, 2, "Antonio", "Costa", "FIXO"],
    [3, 3, "Manuel", "Pereira", "ALA"],
    [4, 4, "Jose", "Santos", "ALA"],
    [5, 5, "Luis", "Ferreira", "PIVOT"],
    [6, 6, "Carlos", "Oliveira", "PIVOT"],
    [7, 7, "Paulo", "Rodrigues", "ALA"],
    [8, 8, "Pedro", "Martins", "ALA"],
    [9, 12, "Miguel", "Gomes", "GK"],
    [10, 10, "Rui", "Fernandes", "ALA"],
    [11, 11, "Tiago", "Goncalves", "ALA"],
    [12, 13, "David", "Almeida", "FIXO"],
    [13, 14, "Ricardo", "Ribeiro", "PIVOT"],
    [14, 15, "Jorge", "Pinto", "ALA"],
    [15, 16, "Daniel", "Carvalho", "ALA"],
    [16, 17, "Andre", "Lopes", "FIXO"]
]

for i, p in enumerate(players, start=14):
    for col, val in enumerate(p, start=1):
        ws.cell(row=i, column=col, value=val)

# Save
filename = "Template_Jogo_Futsal.xlsx"
wb.save(filename)
print(f"Created {filename}")
